lib/console.js

```diff
@@ -367,9 +376,9 @@ Console.prototype.table = function(tabularData, properties) {
 
   const setIter = isSetIterator(tabularData);
   if (setIter)
-    tabularData = previewEntries(tabularData);
+    [ tabularData ] = previewEntries(tabularData);
 
-  const setlike = setIter || isSet(tabularData);
+  const setlike = setIter || (mapIter && !isKeyValue) || isSet(tabularData);
```

Comment:
I am not a fan of this solution. What we should do is to detect the actual type of the `tabularData` before ever overriding that variable. I would just use a second argument to make this easier. As in:

```js
let data = tabularData;

if (mapIter) {
  ...
}
```

Reply:
I’m not sure I understand this comment… Second argument to which function exactly?

Reply:
Sorry, second variable (not argument).

The reason for the issue is that the type can not be properly detected anymore after overriding the original input. So we need a reference to the original input that we never override and one that could potentially be changed. That way the implementation should be easier. I guess I should not have accepted the PR to be landed originally :/

If you want, I can take a stab at this.

Reply:
I mean, we did override `tabularData` before too, so this isn’t really a step back on that (if that’s the variable you’re referring to).

Tbh, I’m still not really getting what you’re trying to say, but it might just be easier to talk about concrete code, yes. :)

> That way the implementation should be easier. I guess I should not have accepted the PR to be landed originally :/

Are you talking about https://github.com/nodejs/node/pull/20719 or the one that introduced the table code in the first place? The former one did not make any changes to this code that go beyond trivial…

Reply:
I mean the original table code in the first place.

Reply:
I am to tired tonight but I am going to have a look at it tomorrow. Is it fine to just push to your branch directly?

Reply:
Yeah, feel free to do that. :)

Reply:
Great :)

Reply:
After looking into this closer it does indeed require some deeper changes for what I thought about.